src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -450,7 +450,17 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         // is what the Terminal did prior to 1.12.
         if (!IsVintageOpacityAvailable())
         {
-            _settings.UseAcrylic(newOpacity < 1.0);
+            const auto oldUseAcrylic{ UseAcrylic() };
+            _runtimeUseAcrylic = newOpacity < 1.0;
+            // If they've changed the value of UseAcrylic, then update the
+            // renderer as well.
+            if (oldUseAcrylic != UseAcrylic() && _renderEngine)
+            {
+                // GH#5098: Inform the engine of the new opacity of the default
+                // text background.
+                auto lock = _terminal->LockForWriting();
+                _renderEngine->SetDefaultTextBackgroundOpacity(UseAcrylic());
+            }

```

Comment:
The other methods just acquire the `_terminal->LockForWriting();` no matter what. Maybe you could do the same here and simplify the code? The call to `SetDefaultTextBackgroundOpacity` itself is very cheap after all...